homework/advanced-calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,137 @@
+#include "advancedCalculator.hpp"
+#include <algorithm>
+#include <stdexcept>
+
+bool is_not_integer(double number) {
+    return std::floor(number) != number;
+}
+
+bool loadNumber(std::string& input, double& number) {
+    size_t the_first_element_of_the_number = 0;
+    std::string set_of_numbers = "1234567890";
+    size_t plus_position = 0;
+    size_t number_position = 0;
+    plus_position = input.find_first_of('+');
+    if (plus_position != std::string::npos) {
+        number_position = input.find_first_of(set_of_numbers);
+        if (plus_position < number_position) {
+            return true;
+        }
+    }
+    try {
+        number = stod(input, &the_first_element_of_the_number);
+        input.erase(0, the_first_element_of_the_number);
+    } catch (std::invalid_argument& e) {
+        return true;
+    }
+    return false;
+}
+
+bool loadCharacter(std::string& input_without_the_first_number, char& operation_symbol) {
+    std::string set_of_characters = "+-*/%!^$";
+    std::string set_of_numbers = "1234567890.";
```

Comment:
same here. You will create and allocate this stirng everytime this is redundant. Bette ruse constexpr char or array